kubernetes/nodecollection.go

```diff
@@ -81,12 +83,24 @@ func downloadNodeData(prefix string,
 		return nil, fmt.Errorf("error occurred requesting container statistics: %v", err)
 	}
 
+	anyNodeReady := false
+
 	for _, n := range nodes.Items {
 		if n.Spec.ProviderID == "" {
 			failedNodeList[n.Name] = errors.New("Provider ID for node does not exist. " +
 				"If this condition persists it will cause inconsistent cluster allocation")
 		}
 
+		// Nodes running in a cluster may not all be ready, check each node individually and if they are not ready
+		// skip the node and create a log.
+		nodeReady := v1Node.IsNodeReady(&n)
+		if !nodeReady {
+			log.Warnf("Node %s was not ready when attempting to get node summary", n.Name)
+			continue
+		} else if anyNodeReady == false {
```

Comment:
Could probably skip this check and just make it true each time. Might help with the cyclomatic complexity lint, too. You could even make it a counter if we wanted to log the number of ready/not ready nodes, but that could be overkill...

Reply:
Ok cool sounds good to me